pumps/stdout.go

```diff
@@ -0,0 +1,109 @@
+package pumps
+
+import (
+	"context"
+	"github.com/TykTechnologies/logrus"
+	"github.com/TykTechnologies/tyk-pump/analytics"
+	"github.com/mitchellh/mapstructure"
+)
+
+var (
+	stdOutPrefix     = "stdout-pump"
+	stdOutDefaultENV = PUMPS_ENV_PREFIX + "_STDOUT" + PUMPS_ENV_META_PREFIX
+)
+
+type StdOutPump struct {
+	CommonPumpConfig
+	conf *StdOutConf
+}
+
+type StdOutConf struct {
+	EnvPrefix    string `mapstructure:"meta_env_prefix"`
+	Format       string `mapstructure:"format"`
+	LogFieldName string `mapstructure:"log_field_name"`
+}
+
+func (s *StdOutPump) GetName() string {
+	return "Stdout Pump"
+}
+
+func (s *StdOutPump) GetEnvPrefix() string {
+	return s.conf.EnvPrefix
+}
+
+func (s *StdOutPump) New() Pump {
+	newPump := StdOutPump{}
+	return &newPump
+}
+
+func (s *StdOutPump) Init(config interface{}) error {
+
+	s.log = log.WithField("prefix", stdOutPrefix)
+
+	s.conf = &StdOutConf{}
+	err := mapstructure.Decode(config, &s.conf)
+
+	if err != nil {
+		s.log.Fatal("Failed to decode configuration: ", err)
+	}
+
+	processPumpEnvVars(s, s.log, s.conf, stdOutDefaultENV)
+
+	if s.conf.Format == "json" {
+		log.Formatter = &logrus.JSONFormatter{}
```

Comment:
@joshblakeley  is this going to change the format of all our logs? I mean, if we configure 2 pumps (one being stdout pump), all the pumps are going to print in json format?

Reply:
Its configured on each pump. My understanding is with pump you need to have them all with the same config unless you have filters anyway because they greedily compete for the same redis key.

Reply:
It seems that when you modify the formatter from `log` (shared between pumps), it modifies the format for every pump
![image](https://user-images.githubusercontent.com/25751713/117311998-1a36fb80-ae5b-11eb-9e50-cfff672dee31.png)

Also, the pumps don't compete for the same key, they all process the same key concurrently.